src/main/java/regression/LR.java

```diff
@@ -0,0 +1,122 @@
+package regression;
+
+import org.apache.commons.math3.stat.regression.SimpleRegression;
+import org.neo4j.graphdb.Entity;
+import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.graphdb.ResourceIterator;
+import org.neo4j.logging.Log;
+import org.neo4j.procedure.*;
+import org.neo4j.procedure.Mode;
+
+import java.io.*;
+import java.util.*;
+import java.util.stream.Stream;
+
+public class LR {
+    @Context
+    public GraphDatabaseService db;
+
+    @Context
+    public Log log;
+
+    @Procedure(value = "regression.linear.create", mode = Mode.READ)
+    public Stream<ModelResult> create(@Name("model") String model) {
+        return Stream.of((new LRModel(model)).asResult());
+    }
+
+    @Procedure(value = "regression.linear.addData", mode = Mode.READ)
+    public Stream<ModelResult> addData(@Name("model") String model, @Name("given") double given, @Name("expected") double expected) {
+        LRModel lrModel = LRModel.from(model);
+        lrModel.add(given, expected);
+        return Stream.of(lrModel.asResult());
+    }
+
+    @Procedure(value = "regression.linear.removeData", mode = Mode.READ)
+    public Stream<ModelResult> removeData(@Name("model") String model, @Name("given") double given, @Name("expected") double expected) {
+        LRModel lrModel = LRModel.from(model);
+        lrModel.removeData(given, expected);
+        return Stream.of(lrModel.asResult());
+    }
+
+    @Procedure(value = "regression.linear.removeModel", mode = Mode.READ)
+    public Stream<ModelResult> removeModel(@Name("model") String model) {
+        return Stream.of(LRModel.removeModel(model));
+    }
+
+    @Procedure(value = "regression.linear.predict", mode = Mode.READ)
+    public Stream<PredictResult> predict(@Name("mode") String model, @Name("given") double given) {
+        LRModel lrModel = LRModel.from(model);
+        return Stream.of(lrModel.predict(given));
+    }
+
+    @Procedure(value = "regression.linear.storeModel", mode = Mode.WRITE)
+    public Stream<ModelResult> storeModel(@Name("model") String model) {
```

Comment:
Yes I agree. This was my original idea, but I ran into issues because I was trying to return byte[] as part of a stream (this is silly because I should have just written a user function). I will fix this, and also convert predict into a function that returns a single value rather than a stream.